src/vs/workbench/contrib/terminal/browser/terminalProcessManager.ts

```diff
@@ -169,6 +172,9 @@ export class TerminalProcessManager extends Disposable implements ITerminalProce
 		} else {
 			this.remoteAuthority = this._workbenchEnvironmentService.remoteAuthority;
 		}
+		const activeWorkspaceRootUri = this._historyService.getLastActiveWorkspaceRoot();
+		this._lastActiveWorkspace = activeWorkspaceRootUri ? withNullAsUndefined(this._workspaceContextService.getWorkspaceFolder(activeWorkspaceRootUri)) : undefined;
+		this._cwdWorkspaceFolder = this._cwdWorkspaceFolder ?? this._lastActiveWorkspace; // fallback to last active workspace if cwd is not available or it is not in workspace
```

Comment:
@Tyriar Last active workspace is not working properly, so getting workspace through cwd instead. Based on https://github.com/microsoft/vscode/pull/179323#discussion_r1160052541 I was expecting it to be the same as cwd.

Reply:
I think i've seen this before too. This looks fine